### PR TITLE
fix(previews): Stop returning true when `getimagesize()` fails

### LINKED ISSUE #45899: [Bug]: Exaggerated presence of error innto the log during syncing of a new device (uplod ndr!)
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

It seems that many old bugs and, probably new ones, have come back to light with Nextcloud 29.0.2 (previous version 29.0 had only troubles with External 
They sometimes cause sync to fail an restarting Nextcloud containers is needed

List of errors found in log viewer during syncing:
- PHP: Optional parameter $userId declared before required parameter $path is implicitly treated as a required parameter at /var/www/html/custom_apps/deck/lib/Service/ConfigService.php#234
- PHP:Trying to access array offset on value of type null at /var/www/html/3rdparty/wapmorgan/mp3info/src/Mp3Info.php#397
- PHP: Undefined array key 4 at /var/www/html/3rdparty/wapmorgan/mp3info/src/Mp3Info.php#397
- PHP: Allowed memory size of 536870912 bytes exhausted (tried to allocate 2097184 bytes) at /var/www/html/3rdparty/guzzlehttp/psr7/src/Stream.php#257
- CRON: DbalException An exception occurred while executing a query: SQLSTATE[HY000]: General error: 7 no connection to the server
- getimagesize(): Corrupt JPEG data: 843 extraneous bytes before marker at /var/www/html/lib/private/legacy/OC_Image.php#697

### Steps to reproduce

1. install from scratch of Docker and Nextcloud AIO 29.0.2.2, then create an account (Ubuntu Debian 22.04 LTS)
2. install the latest Nextcloud Desktop on a Mac
3. synchroniese an already existing exiting folder with thousand of files (documents and images) to the Server
4. wait for errors...

### Expected behavior

Having the whole folder transferred on server without errors...

### Installation method

Community Docker image

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{   
    "system": {
        "one-click-instance": true,
        "one-click-instance.user-limit": 100,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {   
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {   
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "appsallowlist": false,
        "check_data_directory_permissions": false,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "overwritehost": "cloud.lebrit.com",
        "overwriteprotocol": "https",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "cloud.lebrit.com"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "29.0.2.2",
        "overwrite.cli.url": "https:\/\/cloud.lebrit.com\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "updatedirectory": "\/nc-updater",
        "loglevel": 2,
        "app_install_overwrite": [
            "nextcloud-aio"
        ],
        "log_type": "file",
        "logfile": "\/var\/www\/html\/data\/nextcloud.log",
        "log_rotate_size": 10485760,
        "log.condition": {
            "apps": [
                "admin_audit"
            ]
        },
        "preview_max_x": 2048,
        "preview_max_y": 2048,
        "jpeg_quality": 60,
        "enabledPreviewProviders": {
            "1": "OC\\Preview\\Image",
            "2": "OC\\Preview\\MarkDown",
            "3": "OC\\Preview\\MP3",
            "4": "OC\\Preview\\TXT",
            "5": "OC\\Preview\\OpenDocument",
            "6": "OC\\Preview\\Movie",
            "7": "OC\\Preview\\Krita",
            "0": "OC\\Preview\\Imaginary"
        },
        "enable_previews": true,
        "upgrade.disable-web": true,
        "mail_smtpmode": "smtp",
        "trashbin_retention_obligation": "auto, 30",
        "versions_retention_obligation": "auto, 30",
        "activity_expire_days": 30,
        "simpleSignUpLink.shown": false,
        "share_folder": "\/Shared",
        "one-click-instance.link": "https:\/\/nextcloud.com\/all-in-one\/",
        "upgrade.cli-upgrade-link": "https:\/\/github.com\/nextcloud\/all-in-on>
        "maintenance_window_start": 100,
        "allow_local_remote_servers": true,
        "davstorage.request_timeout": 3600,
        "htaccess.RewriteBase": "\/",
        "dbpersistent": false,
        "auth.bruteforce.protection.enabled": true,
        "ratelimit.protection.enabled": true,
        "files_external_allow_create_new_local": true,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "preview_imaginary_url": "***REMOVED SENSITIVE VALUE***",
        "preview_imaginary_key": "***REMOVED SENSITIVE VALUE***",
        "objectstore": {
            "class": "\\OC\\Files\\ObjectStore\\S3",
            "arguments": {
                "bucket": "cloud-lebrit",
                "key": "***REMOVED SENSITIVE VALUE***",
                "secret": "***REMOVED SENSITIVE VALUE***",
                "use_ssl": true,
                "hostname": "it-mil-1.linodeobjects.com"
            }
        }
    }
}
```

### List of activated Apps

```shell
Enabled:
  - admin_audit: 1.19.0
  - calendar: 4.7.6
  - circles: 29.0.0-dev
  - cloud_federation_api: 1.12.0
  - contacts: 6.0.0
  - dashboard: 7.9.0
  - dav: 1.30.1
  - deck: 1.13.0
  - federatedfilesharing: 1.19.0
  - files: 2.1.0
  - files_external: 1.21.0
  - files_pdfviewer: 2.10.0
  - files_reminders: 1.2.0
  - files_sharing: 1.21.0
  - files_trashbin: 1.19.0
  - files_versions: 1.22.0
  - firstrunwizard: 2.18.0
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - nextcloud-aio: 0.5.0
  - nextcloud_announcements: 1.18.0
  - notes: 4.10.0
  - notify_push: 0.6.12
  - oauth2: 1.17.0
  - password_policy: 1.19.0
  - privacy: 1.13.0
  - provisioning_api: 1.19.0
  - recommendations: 2.1.0
  - related_resources: 1.4.0
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - sharebymail: 1.19.0
  - support: 1.12.0
  - systemtags: 1.19.0
  - tasks: 0.16.0
  - text: 3.10.0
  - theming: 2.4.0
  - twofactor_backupcodes: 1.18.0
  - twofactor_totp: 11.0.0-dev
  - user_status: 1.9.0
  - viewer: 2.3.0
  - workflowengine: 2.11.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
See bug description, please.
```

### Additional info

Server Configuration: 
Cpu: 4
RAM: 8 GB
SSD: 160GB
Primary Volume: S3 External Storage
Nextcloud Server: 29.0.2.2  (note: Official Nextcloud AIO)
Desktop macOS: 3.13.0git (build 22490)

### PR DESCRIPTION
**Pull request by joshtrichards**
* Resolves: #45899 

## Summary

Our `checkImageSize()` & `checkImageDataSize()` methods always return `true` if an image is corrupt/invalid[^1] (or unreadable). However, when  `getimagesize()` returns `false` so should `checkImage*()` IMO.

This current behavior causes already detected invalid images to not only bypass the `checkImageMemory()` check (as intended), but to continue to be processed in other ways even though they'll fail.

This impacts all formats, not just jpeg (though most are covered by suppressors at this point).

One could argue the current situation is valid behavior since the intention of the `checkImage*` methods isn't to detect other types of image problems, so it's a bit of an overloaded use. But in all cases I can conceive of, other image manipulations are going to fail anyhow if `getimagesize()` does. And we're immediately calling `getimagesize()` once again in many cases.

P.S. With the fix to `checkImageSize()`, the line with the suppressor shouldn't even be executed any longer. The suppressor added here is just for good measure and to be consistent with the other image formats. If the behavior of the `checkImage*()` methods is desired to be retained as-is, I can eliminate those changes and keep only the suppressor.

[^1]: (or at least corrupt enough their image data can't be retrieved by gd) 

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by joshtrichards** [comment]
Thinking more about this, we should probably just use the suppressor and not over-use these other methods...

File: 3rdparty
Comment (artonge) [inline 3rdparty:1]: Is this change on purpose?

Comment (joshtrichards) [reply 3rdparty:1]: No.

Comment (joshtrichards) [reply 3rdparty:1]: Fixed.

**Review by artonge — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]
